extensions/amp-auto-ads/0.1/anchor-ad-strategy.js

```diff
@@ -77,12 +91,25 @@ export class AnchorAdStrategy {
     if (!optInStatus) {
       return false;
     }
-    for (let i = 0; i < optInStatus.length; i++) {
-      if (optInStatus[i] == OPT_IN_STATUS_ANCHOR_ADS) {
-        return true;
-      }
+    return optInStatus.some(
```

Comment:
The logic for whether to do sticky ad and whether to fill it is spread over multiple functions in the code with this setup. This makes it a bit harder to follow and also potentially more susceptible to bugs if someone were to edit one part but not the other. E.g. if someone tried to disable to no-fill functionality by making hasOptInStatusNoFillAnchorAds_ always return false, then with the current code they would actually just cause filled anchor ads to be inserted instead.

I think it would be cleaner to try and keep all this decision logic in a single function. You could have two functions:
isFilledAnchorAdEnabled_() -> just checks for OPT_IN_STATUS_ANCHOR_ADS
isNoFillAnchorAdEnabled_() -> just checks for OPT_IN_STATUS_NO_FILL_ANCHOR_ADS

Then the logic in run() becomes:
run() {
  const filledAnchorEnabled = this.isFilledAnchorAdEnabled_();
  const noFillAnchorEnabled = this.isNoFillAnchorAdEnabled_();

  if (!filledAnchorEnabled && !noFillAnchorEnabled) {
    return Promise.resolve(false);
  }
  ...
  this.placeStickyAd_(noFillAnchorEnabled && !filledAnchorEnabled);
}

And the placeStickyAd_ method would take an argument called noFill:
placeStickyAd_(noFill) {
  ...
}

Then all the logic to decide when anchor should run and whether it fills or not is in one single place in the code.

Reply:
Done. Thanks!!